### DISCUSSION TITLE
Explicit Defaults and Override intent for paramsRef in Gateway and GatewayClass

### DISCUSSION THREAD

**Discussion by arkodg** [kubernetes-sigs/gateway-api#2154] [General]
https://gateway-api.sigs.k8s.io/geps/gep-1867/ introduces `paramsRef` at the `Gateway` level, similar to what exists in the `GatewayClass` as well, allowing CRDs to be now linked at both these levels.
The GEP does mention that the `paramsRef` in `GatewayClass` is expected to be treated as default implying that if `paramsRef` is set in `Gateway`, its value MUST always be used instead.

Can this intent be specified using an API field instead under `infrastructure` in `GatewayClass`  ?
e.g. such as `bool override` which defaults to `false` allowing the `GatewayClass`'s `paramsRef` to be treated as a default, and if set to `true`, the `paramsRef` value in `GatewayClass` overrides all the values in the child `Gateway`'s `paramRef` setting.

**Comment by youngnick** [comment]
Coming back to this, I'm reluctant to do this initially - this would be pretty complicated and require a lot of separated testing (since we'll need to test `override` `true` and `false`). Right now (in provisional and implementable), we can get away with limited testing.

That said, making a decision here should be a requirement for `infrastructure` to go to Standard.

**Reply by youngnick** [reply]
This probably needs an update to GEP-1867 to capture it as future work/graduation criteria, I'll add a todo for myself.
